lib/eco/incparser/incparser.py

```diff
@@ -784,15 +784,28 @@ def interpret_annotation(self, node, production):
                 node.alternate = astnode
 
     def is_reusable_astnode(self, old, new):
-        from grammar_parser.bootstrap import AstNode
-        if type(old) is not AstNode or type(new) is not AstNode:
+        from grammar_parser.bootstrap import AstNode, ListNode
+
+        if type(old) is not type(new):
             return False
+
         if old.name != new.name:
             return False
-        for key in old.children:
-            if old.children.get(key) is not new.children.get(key):
+
+        if type(old) is ListNode:
```

Comment:
What are the possible types of `old`. Is it just `ListNode` and `AstNode` or other things too?

Reply:
It's either `AstNode`, `ListNode`, or terminals and nonterminals.

Reply:
OK, got it.